lib/addons/src/hooks/useCallbackRef.ts

```diff
@@ -0,0 +1,14 @@
+import { useState, useCallback, RefObject } from 'react';
+
+// @TODO should be React.MutableRefObject....
+export function useCallbackRef<T extends HTMLElement = HTMLElement>() {
+  const [ref, setRef] = useState<RefObject<T | null>>(null);
+  const fn = useCallback(
+    (node: RefObject<T | null>) => {
+      setRef(node);
+    },
+    [setRef]
+  );
+
+  return { ref, fn };
+}
```

Comment:
Can you explain to me what the point of this is?

Reply:
Not exactly as in "is this absolutely necessary" - it seems it just makes it easier to have a cleaner output from the hook useDOMRect as the useEffect that uses the setBounds gets called one time less

This is as I remember it, this is code I copied some time ago from somewhere and tweaked :-D

If you can see we can avoid it and not have extra render due to useDOMRect having setBounds called once too many in the start then yes - lets get rid of it!

Reply:
I will use `useLayoutEffect` and now this is not required anymore

Reply:
Might have to be required, will have to start coding against a use case again

React has the type RefCallback made for this situation it seems, and other hooks I have been looking at that does the same also employes a callback for ref in one variant or another ([like here](https://github.com/ZeeCoder/use-resize-observer/blob/master/src/index.ts))

Reply:
> I will use `useLayoutEffect` and now this is not required anymore

won't work for delayed mounts - will need the callback